diff -ur tristripper-1.1.0-beta-5o/include/detail/connectivity_graph.h tristripper-1.1.0-beta-5n/include/detail/connectivity_graph.h
--- tristripper-1.1.0-beta-5o/include/detail/connectivity_graph.h	2005-06-08 20:47:26.000000000 -0600
+++ tristripper-1.1.0-beta-5n/include/detail/connectivity_graph.h	2008-02-21 19:47:44.000000000 -0700
@@ -24,7 +24,7 @@
 	namespace detail
 	{
 
-		void make_connectivity_graph(graph_array<triangle> & Triangles, const indices & Indices);
+		void make_connectivity_graph(graph_array<triangle> & Triangles, const index * Indices, size_t Index_Count);
 
 	}
 
diff -ur tristripper-1.1.0-beta-5o/include/tri_stripper.h tristripper-1.1.0-beta-5n/include/tri_stripper.h
--- tristripper-1.1.0-beta-5o/include/tri_stripper.h	2005-06-08 20:47:26.000000000 -0600
+++ tristripper-1.1.0-beta-5n/include/tri_stripper.h	2008-02-21 19:48:30.000000000 -0700
@@ -69,7 +69,7 @@
 {
 public:
 
-	tri_stripper(const indices & TriIndices);
+	tri_stripper(const index* TriIndices, size_t count);
 
 	void Strip(primitive_vector * out_pPrimitivesVector);
 
diff -ur tristripper-1.1.0-beta-5o/src/connectivity_graph.cpp tristripper-1.1.0-beta-5n/src/connectivity_graph.cpp
--- tristripper-1.1.0-beta-5o/src/connectivity_graph.cpp	2005-06-08 20:47:26.000000000 -0600
+++ tristripper-1.1.0-beta-5n/src/connectivity_graph.cpp	2008-02-21 19:49:15.000000000 -0700
@@ -54,9 +54,9 @@
 
 
 
-void make_connectivity_graph(graph_array<triangle> & Triangles, const indices & Indices)
+void make_connectivity_graph(graph_array<triangle> & Triangles, const index* Indices, size_t Index_Count)
 {
-	assert(Triangles.size() == (Indices.size() / 3));
+	assert(Triangles.size() == (Index_Count / 3));
 
 	// Fill the triangle data
 	for (size_t i = 0; i < Triangles.size(); ++i)
diff -ur tristripper-1.1.0-beta-5o/src/tri_stripper.cpp tristripper-1.1.0-beta-5n/src/tri_stripper.cpp
--- tristripper-1.1.0-beta-5o/src/tri_stripper.cpp	2005-06-08 20:47:26.000000000 -0600
+++ tristripper-1.1.0-beta-5n/src/tri_stripper.cpp	2008-02-21 19:50:47.000000000 -0700
@@ -24,8 +24,8 @@
 
 
 
-tri_stripper::tri_stripper(const indices & TriIndices)
-	: m_Triangles(TriIndices.size() / 3), // Silently ignore extra indices if (Indices.size() % 3 != 0)
+tri_stripper::tri_stripper(const index* TriIndices, size_t count)
+	: m_Triangles(count / 3), // Silently ignore extra indices if (Indices.size() % 3 != 0)
 	  m_StripID(0),
 	  m_FirstRun(true)
 {
@@ -34,7 +34,7 @@
 	SetBackwardSearch();
 	SetPushCacheHits();
 
-	make_connectivity_graph(m_Triangles, TriIndices);
+	make_connectivity_graph(m_Triangles, TriIndices, count);
 }
 
 
